jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryActor.kt

```diff
@@ -0,0 +1,189 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.insights
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Disposer
+import com.intellij.ui.table.TableView
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient
+import software.amazon.awssdk.services.cloudwatchlogs.model.GetQueryResultsRequest
+import software.amazon.awssdk.services.cloudwatchlogs.model.ResourceNotFoundException
+import software.amazon.awssdk.services.cloudwatchlogs.model.ResultField
+import software.aws.toolkits.core.utils.error
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.jetbrains.utils.getCoroutineUiContext
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+import software.aws.toolkits.resources.message
+
+sealed class QueryActor<T>(
+    private val project: Project,
+    protected val client: CloudWatchLogsClient,
+    private val table: TableView<T>
+) : CoroutineScope by ApplicationThreadPoolScope("InsightsResultTable"), Disposable {
+    val channel = Channel<MessageLoadQueryResults>()
+    protected var moreResultsAvailable: Boolean = false
+    protected abstract val emptyText: String
+    protected abstract val tableErrorMessage: String
+    protected abstract val notFoundText: String
+
+    private val edtContext = getCoroutineUiContext(disposable = this@QueryActor)
+    private val exceptionHandler = CoroutineExceptionHandler { _, e ->
+        LOG.error(e) { "Exception thrown in Query Results not handled:" }
+        notifyError(title = message("general.unknown_error"), project = project)
+        table.setPaintBusy(false)
+        Disposer.dispose(this)
```

Comment:
We should probably not do this, can you change it to `channel.close()` then fix the other instance of this